lib/cli.ts

```diff
@@ -0,0 +1,159 @@
+ * It takes care of parsing command line options.
+ *
+ * Values from command line options override values from the config.
+ */
+
+var args: Array<string> = [];
+
+process.argv.slice(2).forEach(function(arg: string) {
+  var flag: string = arg.split('=')[0];
+
+  switch (flag) {
+    case 'debug':
+      args.push('--nodeDebug');
+      args.push('true');
+      break;
+    case '-d':
+    case '--debug':
+    case '--debug-brk':
+      args.push('--v8Debug');
+      args.push('true');
+      break;
+    default:
+      args.push(arg);
+      break;
+  }
+});
+
+export interface ProtractorArgs {
+  _: Array<string>;
+  help?: boolean;
+  version?: boolean;
+  capabilities?: Object;
+  specs?: string|Array<string>;
+  exclude?: string|Array<string>;
+  elementExplorer?: boolean;
+}
+;
+
+optimist
+    .usage(
+        'Usage: protractor [configFile] [options]\n' +
+        'configFile defaults to protractor.conf.js\n' +
+        'The [options] object will override values from the config file.\n' +
+        'See the reference config for a full list of options.')
+    .describe('help', 'Print Protractor help menu')
+    .describe('version', 'Print Protractor version')
+    .describe('browser', 'Browsername, e.g. chrome or firefox')
+    .describe('seleniumAddress', 'A running selenium address to use')
+    .describe('seleniumSessionId', 'Attaching an existing session id')
+    .describe(
+        'seleniumServerJar', 'Location of the standalone selenium jar file')
+    .describe(
+        'seleniumPort', 'Optional port for the selenium standalone server')
+    .describe('baseUrl', 'URL to prepend to all relative paths')
+    .describe('rootElement', 'Element housing ng-app, if not html or body')
+    .describe('specs', 'Comma-separated list of files to test')
+    .describe('exclude', 'Comma-separated list of files to exclude')
+    .describe('verbose', 'Print full spec names')
+    .describe('stackTrace', 'Print stack trace on error')
+    .describe('params', 'Param object to be passed to the tests')
+    .describe('framework', 'Test framework to use: jasmine, mocha, or custom')
+    .describe('resultJsonOutputFile', 'Path to save JSON test result')
+    .describe('troubleshoot', 'Turn on troubleshooting output')
+    .describe('elementExplorer', 'Interactively test Protractor commands')
+    .describe(
+        'debuggerServerPort',
+        'Start a debugger server at specified port instead of repl')
+    .alias('browser', 'capabilities.browserName')
+    .alias('name', 'capabilities.name')
+    .alias('platform', 'capabilities.platform')
+    .alias('platform-version', 'capabilities.version')
+    .alias('tags', 'capabilities.tags')
+    .alias('build', 'capabilities.build')
+    .alias('grep', 'jasmineNodeOpts.grep')
+    .alias('invert-grep', 'jasmineNodeOpts.invertGrep')
+    .alias('explorer', 'elementExplorer')
+    .string('capabilities.tunnel-identifier')
+    .check(function(arg: ProtractorArgs) {
```

Comment:
It definitely isn't an array of strings.  I didn't change the `if` statement on the line below.  Maybe it's just `{_: string[]}` though